product-scenarios/1-integrating-systems-that-communicate-in-heterogeneous-message-formats/1.6-xml-message-enrichment/1.6.3-Replace-a-value-of-the-incoming-message-using-enrich-mediator/src/test/java/org/wso2/carbon/ei/scenario/test/ReplaceElementsTest.java

```diff
@@ -0,0 +1,199 @@
+import org.wso2.carbon.esb.scenario.test.common.ScenarioTestBase;
+import org.wso2.carbon.esb.scenario.test.common.http.HTTPUtils;
+
+import java.io.IOException;
+import javax.xml.stream.XMLStreamException;
+
+/**
+ * This class is to test if xml payload can be enriched before it goes to the backend server by replacing elements
+ * in the payload. This can be done using enrich mediator. This class focusses on various ways of achieving that.
+ */
+public class ReplaceElementsTest extends ScenarioTestBase {
+
+    @BeforeClass(alwaysRun = true)
+    public void init() throws Exception {
+        super.init();
+    }
+
+    //This test is to verify if payload can be modified by replacing the body of payload using a payload stored in a property
+    @Test(description = "1.6.3.1 - Replacing body of payload using a payload stored in a property")
+    public void ReplaceMessageBodyUsingPayloadStoredInProperty() throws IOException, XMLStreamException {
+        String url = getProxyServiceURLHttp("1_6_3_1_Proxy_replace_messageBody_usingPayloadStoredInProperty");
+
+        String request =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                        + "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:sam=\"http://sample.wso2.org\" xmlns:xsd=\"http://sample.wso2.org/xsd\">\n"
+                        + "   <soapenv:Body>\n"
+                        + "      <sam:placeOrder>\n"
+                        + "         <sam:order>\n"
+                        + "            <xsd:price>12</xsd:price>\n"
+                        + "            <xsd:productid>IC002</xsd:productid>\n"
+                        + "            <xsd:quantity>2</xsd:quantity>\n"
+                        + "            <xsd:reference>ref</xsd:reference>\n"
+                        + "         </sam:order>\n"
+                        + "      </sam:placeOrder>\n"
+                        + "   </soapenv:Body>\n"
+                        + "</soapenv:Envelope>";
+
+        String expectedResponse =
+                "<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n"
+                        + "<soapenv:Envelope xmlns:soapenv=\"http://schemas.xmlsoap.org/soap/envelope/\" xmlns:sam=\"http://sample.wso2.org\" xmlns:xsd=\"http://sample.wso2.org/xsd\">\n"
+                        + "   <soapenv:Body>\n"
+                        + "      <sam:order>\n"
+                        + "         <xsd:price>12</xsd:price>\n"
+                        + "         <xsd:productid>IC002</xsd:productid>\n"
+                        + "         <xsd:quantity>2</xsd:quantity>\n"
+                        + "         <xsd:reference>ref</xsd:reference>\n"
+                        + "      </sam:order>\n"
+                        + "   </soapenv:Body>\n"
+                        + "</soapenv:Envelope>";
+
+        HTTPUtils.invokeSoapActionAndAssert(url, request, ScenarioConstants.MESSAGE_ID, expectedResponse, 200,
```

Comment:
Let's add test case ID for message ID header

Reply:
This is done.